services/provider/server/server.go

```diff
@@ -206,3 +212,121 @@ func Start(port int, providerServer *OCSProviderServer, opts []grpc.ServerOption
 		klog.Fatalf("failed to start gRPC server: %v", err)
 	}
 }
+
+func (s *OCSProviderServer) getJSONConnectionDetails(consumerResource *ocsv1alpha1.StorageConsumer) ([]externalSC.ExternalResource, error) {
+	var j []externalSC.ExternalResource
+
+	// Configmap with mon endpoints
+	configmap, err := s.consumerAPI.GetConfigMap("rook-ceph-mon-endpoints", s.namespace)
+	if err != nil {
+		return []externalSC.ExternalResource{}, fmt.Errorf("failed to get `rook-ceph-mon-endpoints` configMap. Error %v", err)
+	}
+
+	cmData := strings.Split(configmap.Data["data"], ",")
+	data := map[string]string{
+		"data":     cmData[0],
+		"maxMonId": "0",
+		"mapping":  "{}",
+	}
+	name := "rook-ceph-mon-endpoints"
+	kind := "ConfigMap"
+
+	j = append(j, externalSC.ExternalResource{Name: name, Kind: kind, Data: data})
+
+	// Secret storing cluster mon.admin key, fsid and name
+	scMon, err := s.consumerAPI.GetSecret("rook-ceph-mon", s.namespace)
+	if err != nil {
+		return []externalSC.ExternalResource{}, fmt.Errorf("failed to get `rook-ceph-mon` secret. Error %v", err)
+	}
+	data = map[string]string{
+		"fsid":         string(scMon.Data["fsid"]),
+		"mon-secret":   "mon-secret",
+		"admin-secret": "admin-secret",
+	}
+	name = "rook-ceph-mon"
+	kind = "Secret"
+
+	j = append(j, externalSC.ExternalResource{Name: name, Kind: kind, Data: data})
+
+	// Ceph user secret
+	storageConsumer, err := s.consumerAPI.GetStorageConsumer(consumerResource.Name, s.namespace)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get storageConsumer resource for the given consumer: %v", err)
+	}
+	cephUser := storageConsumer.Status.CephObjects.CephUser
+	scCephUser, err := s.consumerAPI.GetSecret(fmt.Sprintf("rook-ceph-client-%s", cephUser), s.namespace)
+	if err != nil {
+		return []externalSC.ExternalResource{}, fmt.Errorf("failed to get `rook-ceph-client %s` secret. Error %v", "", err)
```

Comment:
why is `%s` pointing to an emtpty string?

Reply:
added string var.